src/VisualStudio/Core/Def/Implementation/TaskList/ExternalErrorDiagnosticUpdateSource.cs

```diff
@@ -420,26 +420,29 @@ public void Done()
 
             public bool SupportedDiagnosticId(ProjectId projectId, string id)
             {
-                if (_diagnosticIdMap.TryGetValue(projectId, out var ids))
+                lock (_diagnosticIdMap)
                 {
-                    return ids.Contains(id);
-                }
+                    if (_diagnosticIdMap.TryGetValue(projectId, out var ids))
```

Comment:
usage scenario of MultiDictionary is when a code path can have mostly 1 item but sometime multiple items. and code is optimized to handle those case efficiently. which is not this case. I dont think that is good one to reuse in this case.